### ISSUE TITLE
✅ 🐞 Bug with anchors when table of contents is made sticky

### ISSUE THREAD

**Issue by mediascite** [elementor/elementor#15620]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

When the Table of Content is used inside a sticky section or is made sticky it does not point correctly to the anchors.

## Steps to reproduce

- Create a section with 2 columns.
- On the right add a text widget and add text with some titles (4 or 5 h5 titles for example).
- On the left one add a table of contents. Make it sticky.

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
Operating System: Linux
Software: Apache/2.4.41 (Ubuntu)
MySQL version: (Ubuntu) v8.0.23
PHP Version: 7.4.3
PHP Max Input Vars: 1000
PHP Max Post Size: 64M
GD Installed: Yes
ZIP Installed: Yes
Write Permissions: All right
Elementor Library: Connected

Wordpress info
Version: 5.7.2
WP Multisite: No
Max Upload Size: 64 Mo
Memory limit: 128M

Theme
Name: Twenty Twenty-One
Version: 1.3
Author: the WordPress team
Child Theme: No

Elementor
Version: 3.3.0

Elementor Pro
Version: 3.3.2

</details>

**Comment by Aviad-Herman** [comment]
Sorry, couldn't reproduce using these steps.
What browser are you using?
A video would be great.

In any case, I would suggest contacting support because this doesn't seem to be a general cross-system problem.

**Comment by mediascite** [comment]
Hi,

i have contacted support and they answer that the problem was known... But they can't answer when it will be resolved.
You can see the problem here : https://bit.ly/3i2ucJ0
The first anchor isn't active when you load the page

**Comment by Aviad-Herman** [comment]
https://www.loom.com/share/fc51cda7c2d04bb1a1326038d7977bc3

Is the problem only on specific browsers / operating systems? It seems to be working fine in the video. Perhaps I don't understand the problem.

**Comment by mediascite** [comment]
The first link isn't marked as active when you load the page (it is on the second one).
You can see the problem here too : https://onco-nouvelle-aquitaine.fr/covid-19-cancer/informations-covid-19-et-cancer/
and here : https://onco-nouvelle-aquitaine.fr/prises-en-charge/cancers-rares/
The first title isn't selected when you load the page. And when you navigate on the page, sometimes anchor selected is not the good one
video here : https://www.loom.com/share/ae2610c2684d4dd192178b8600d69f19

**Comment by Aviad-Herman** [comment]
Perfect explanation, thanks!
I can't promise exact dates but I will try to promote this internally.

**Comment by iStuff0** [comment]
Yes, it would be nice if they fixed the TOC widget as it's been like this for as long as i've used Elementor. 

It also bounces around like mad sometimes, pretty unprofessional looking: https://www.loom.com/share/bdac403f20064fcca6ebbda965005ec9

**Comment by masoudahmadpour** [comment]
+1

**Comment by mediascite** [comment]
Hi,
this bug is really annoying and it seems that nothing is done to resolve it.
Could you resolve it for the next release please ? Thanks !

**Comment by lrdqs** [comment]
It's been a year now. Table of contents active item highlighting is still bugged, jumping around. When scrolling, an item will be highlighted as soon as it enters the screen from the bottom, scrolling further will highlight the previous item in the list, then jumping to the next item entering the screen.

**Comment by sermalefico** [comment]
No works when is sticky

**Comment by voyagedigital** [comment]
This is still a problem, why has the dev team not fixed this by now? Year and a half should be plenty of time?

**Comment by bennyqp** [comment]
+1

**Comment by RudiVL** [comment]
My gosh @rami-elementor @Aviad-Herman and the rest... How can you be so rude to not come back to this topic? 
It is a very simple error... and you got all the info to solve it.
I have exactly the same problem. Just a TOC in a column next to my main text/article. Made the TOC sticky and from that point it just does not work anymore. It will go a bit down, but where the div where the TOC is in, touches the point where it should go sticky, it will not move anymore. 

Why, for software where we pay so much money for, you are just not for years coming back at problems like this?
Have seen the 2023 talk, it all seemed like you would really get much closer to users, because in all fairness you have been ignoring them a long time... For years I am customer, Years had fun but my gosh have I also had problems that never got solved. 

Please take this seriously and get to work on this problem ok?

**Comment by marowi85** [comment]
I am facing the same issue.

We have a two column layout. TOC widget in left column and sticky. The content in the right column.

The TOC widget does not correctly refer to the anchors and the highlighted links are also misleading.

**Comment by Michaelgimii** [comment]
H @nicholaszein, Are you aware of this bug?

**Comment by chris1511** [comment]
Hello, I just faced the same problem.
I managed to solve it by putting a z-index:auto; to the table of contents in the custom CSS, then regenerating the Elementor files in Elementor/Tools and clearing the cache.
Maybe it will work for you.😉

**Comment by voyagedigital** [comment]
Flexbox containers have fixed this. So use the new containers and it will work 👌🏻

**Comment by narnyworld** [comment]
I tried to convert it to Flexbox containers - the same thing happens. I have two columns (two containers) and on the left one I have a sticky to the top, table of content. The selector is still jumping back and forth.

**Comment by Paul-Heron** [comment]
There is still an issue when using containers. I have a brand new site with the same issue and it has used containers since the start.

**Comment by voyagedigital** [comment]
That's strange, I built this page using the same method, but with flexbox containers and it works as expected: https://geostruct.com/method/#plan

**Comment by JabeztheStoic** [comment]
Elementor Support ... I love you but ... why don't you F**king respond to tiny issues that make big impacts? Like this one.

**Comment by manuel-will** [comment]
This still needs fixing +1

**Comment by rizio** [comment]
I'm still seeing the same bug/issue here with this. I have two containers, in one container the TOC. I have the "Container" flag set correctly. The TOC is in a container that's sticky and the links don't work.

**Comment by samuelmente** [comment]
+1 The issue is still there. Any updates on a fix?

**Comment by layscough** [comment]
+1 Still broken for me. On page load, second item in the TOC is highlighted and highlight changes to the next item as soon as it comes into viewport.

**Comment by rubenoam** [comment]
+1 Issue still not fixed

**Comment by tribitas** [comment]
Issue is still broken, can confirm

**Comment by PERMI-Design** [comment]
+1

**Comment by tadytomas** [comment]
Same issue here. Any updates, Elementor?

**Comment by mrthanasis** [comment]
+1

**Comment by ts3once** [comment]
+1

**Comment by frydhogg** [comment]
+1 please fix it

**Comment by Duuble-U** [comment]
+1

**Comment by nicholaszein** [comment]
Hi everyone!

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro 3.25.0! 🥳 

This version introduced a new system for handling anchors and smooth-scroll, and included a new **`Anchor Offset`** control under the **`Sticky`** settings that allows to set a proper offset from the top of the page for anchor links when targeted.

You can learn more about it here:
https://elementor.com/help/sticky-headers/

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

> [!NOTE]
> If you think this is still an issue caused by Elementor, feel free to open a new 🐞 [Bug Report](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml) with detailed information so we investigate your issue better.

Cheers 🥂
